stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/IdGenerationDomainTest.kt

```diff
@@ -1,69 +1,66 @@
 package io.getstream.chat.android.livedata
 
-import androidx.test.ext.junit.runners.AndroidJUnit4
+import android.content.Context
+import android.os.Handler
 import com.google.common.truth.Truth
-import io.getstream.chat.android.client.api.models.QuerySort
-import io.getstream.chat.android.client.models.Channel
-import io.getstream.chat.android.client.models.Filters
-import io.getstream.chat.android.livedata.controller.QueryChannelsSpec
-import org.junit.Test
-import org.junit.runner.RunWith
+import com.nhaarman.mockitokotlin2.mock
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.models.User
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
 
-@RunWith(AndroidJUnit4::class)
-internal class IdGenerationDomainTest : BaseDomainTest() {
+internal class IdGenerationDomainTest {
+
+    private lateinit var chatDomainImpl: ChatDomainImpl
+    private lateinit var currentUserFake: User
+
+    @BeforeEach
+    fun init() {
+        val contextMock = mock<Context>()
+        val clientMock = mock<ChatClient>()
+        val handlerFake = Handler()
+        chatDomainImpl = ChatDomainImpl(
+            appContext = contextMock,
+            client = clientMock,
+            handler = handlerFake,
+            backgroundSyncEnabled = true,
+            offlineEnabled = true,
+            recoveryEnabled = false,
+            userPresence = true,
+        )
+        currentUserFake = randomUser()
+    }
+
+    @Test
+    fun `Given ChatDomainImpl with no currentUser When generateMessageId() called Should throw`() {
+        assertThrows<UninitializedPropertyAccessException> { chatDomainImpl.currentUser }
+        assertThrows<UninitializedPropertyAccessException> { chatDomainImpl.generateMessageId() }
+    }
 
     @Test
-    fun messageIdGeneration() {
+    fun `Given ChatDomainImpl with valid currentUser When generateMessageId() called Should return not-empty messageId`() {
+        setCurrentUser()
+
         val messageId = chatDomainImpl.generateMessageId()
 
-        Truth.assertThat(messageId).isNotNull()
         Truth.assertThat(messageId).isNotEmpty()
     }
 
     @Test
-    fun messageIdGenerationIsUnique() {
+    fun `Given ChatDomainImpl with valid currentUser When generateMessageId() called Should return unique messageId`() {
+        setCurrentUser()
+
         val idMap = sortedSetOf<String>()
-        for (x in 0..100) {
+        (0..1000000).forEach {
             val messageId = chatDomainImpl.generateMessageId()
             Truth.assertThat(idMap).doesNotContain(messageId)
             idMap.add(messageId)
         }
     }
 
-    @Test
```

Comment:
This method is moved to another test class because it's totally different stuff than `ChatDomainImpl`.